api/xds_protocol.rst

```diff
@@ -36,43 +36,122 @@ occurs.
+even if the 99 that were already subscribed to have not changed (in LDS/CDS). This mechanism can be a scalability limitation,
+which is why the incremental API variant was introduced. The incremental approach allows both the client and server to indicate
+only deltas relative to their previous state -- i.e., the client can say that it wants to add or remove its subscription to a
+particular resource name without resending those that have not changed, and the server can send updates only for those
+resources that have changed. For details on the incremental protocol, see :ref:`Incremental xDS <xds_protocol_incremental>`
+below.
+
+The second dimension is using a separate gRPC stream for each resource type vs. aggregating all resource types onto a
+single gRPC stream. The former approach was the original mechanism used by xDS, and it offers an eventual consistency
+model. The latter approach was added for environments in which explicit control of sequencing is required. For details, see
+:ref:`Eventual consistency considerations <xds_protocol_eventual_consistency>` below.
+
+So, the four variants of the xDS API are:
+
+1. State of the World (Basic xDS): SotW, separate gRPC stream for each resource type
+2. Incremental xDS: incremental, separate gRPC stream for each resource type
+3. Aggregated Discovery Service (ADS): SotW, aggregate stream for all resource types
+4. Incremental ADS: incremental, aggregate stream for all resource types
+
+In general, new xDS implementations are encouraged to use the last variant.
+
+RPC Services and Methods for Each Variant
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+For the non-aggregated API variants, there is a separate RPC service for each resource type. Each of these RPC services can
+provide a method for each of the SotW and Incremental API variants. Here are the RPC services and methods for each resource
+type:
+
+-  Listener: Listener Discovery Service (LDS)
+   -  SotW: ListenerDiscoveryService.StreamListeners
+   -  Incremental: ListenerDiscoveryService.DeltaListeners
+-  RouteConfiguration: Route Discovery Service (RDS)
+   -  SotW: RouteDiscoveryService.StreamRoutes
+   -  Incremental: RouteDiscoveryService.DeltaRoutes
+-  VirtualHost: Virtual Host Discovery Service (VHDS)
+   -  SotW: N/A
+   -  Incremental: VirtualHostDiscoveryService.DeltaVirtualHosts
+-  Cluster: Cluster Discovery Service (CDS)
+   -  SotW: ClusterDiscoveryService.StreamClusters
+   -  Incremental: ClusterDiscoveryService.DeltaClusters
+-  ClusterLoadAssignment: Endpoint Discovery Service (EDS)
+   -  SotW: EndpointDiscoveryService.StreamEndpoints
+   -  Incremental: EndpointDiscoveryService.DeltaEndpoints
+-  Secret: Secret Discovery Service (SDS)
+   -  SotW: SecretDiscoveryService.StreamSecrets
+   -  Incremental: SecretDiscoveryService.DeltaSecrets
+-  Runtime: Runtime Discovery Service (RTDS)
+   -  SotW: RuntimeDiscoveryService.StreamRuntime
+   -  Incremental: RuntimeDiscoveryService.DeltaRuntime
```

Comment:
One lingering piece of technical debt is rendering the service definitions via `protodoc`. I think we would want it to look something like https://www.envoyproxy.io/docs/envoy/latest/operations/admin treatment of REST operations. I wonder if you can do something with RST to make it look like that here.

Reply:
Fair enough, it's fine to punt this to later.